### PR TITLE
fixed issue 7818 - informing users about sequential order of AirCRaft dataset, suggesting to shuffle the dataloader

### PR DESCRIPTION
**Pull request by EyalMichaeli**
fixes issue #7818

### CODE REVIEW

File: torchvision/datasets/fgvc_aircraft.py [hunk 1]
```diff
@@ -112,3 +114,21 @@ def _download(self) -> None:
...
+    Print a one-time warning for the user that in order to avoid the probable bias and performace degragation, the dataloader needs to be shuffled.
+    Check if a warning was already printed, and if not, print it. Saves a boolean file in the torchvision cache directory.
+    """
+    cache_dir = os.path.join(os.path.expanduser('~'), '.cache', 'torch', 'hub', "FGVCAircraft")
+    boolean_file_path = os.path.join(cache_dir, "FGVCAircraft_shuffle_warning.txt")
+
+    if not os.path.exists(boolean_file_path):
+        print("Important: The samples in this dataset are organized sequentially by class.")
+        print("To ensure unbiased performance, it is recommended to shuffle the dataset before use.")
+        print("To do so, you can set the 'shuffle' argument in the dataloader to 'True'.")
```
Comment (NicolasHug) [inline torchvision/datasets/fgvc_aircraft.py:131]: thanks for the PR @EyalMichaeli . As detailed in https://github.com/pytorch/vision/issues/7818#issuecomment-1672786974, we won't be able to merge this as we don't plan on raising warnings for unshuffled datasets.

Just a minor note, hoping that helps: for warnings we don't use `print`, we use `warnings.warn` which can be filtered in specific ways and also avoids warning twice (thus removing the need for the boolean file you created).

**Comment by EyalMichaeli** [comment]
Thanks for the note. 
As per the data loader docs
https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader
shuffle is False by default. 

I used a repo which didn’t use shuffle in the data loader, and when I ran it on the AirCraft dataset, I got strange and much worse results. Only when I debugged it and saw that the problem is sequential order in the labels, I was able to fix it.
Just wanted to make sure no one will waste time on this in the future.
Up to you guys 🙂

**Comment by NicolasHug** [comment]
> shuffle is False by default.

You're right, sorry.

> I used a repo which didn’t use shuffle in the data loader

It is absolutely something that repo should fix. Datasets come in all shapes and forms and we shouldn't assume that they are already shuffled.

Even if datasets were shuffled, imagine they were just stored in a pre-shuffled format that doesn't depend on any RNG such that the "shuffling" is always the same: you think you're getting random data but in reality you're getting the same all the time, and that can make a lot of experiment lose statistical significance. It's best to never assume anything and always shuffle manually. This way, users have full control on what they want to do.

I agree it can be a bit surprising / frustrating though. I stumbled on similar things myself in the past (things are much much worse with iterable datasets for example).

**Comment by EyalMichaeli** [comment]
I can understand it. 
I wish that at least the data loader was shuffling by default, at least. 
Anyway, closing the issue.
Keep up the good work 🙂
